web-frontend/src/pages/Settings/SettingsSignRule.tsx

```diff
@@ -19,36 +28,57 @@ interface SettingsSignRulesProps {
 export default function SettingsSignRules(props: SettingsSignRulesProps) {
   const { nickname } = props;
 
-  const {
-    data: config,
-    isLoading: isConfigLoading,
-    error: configError,
-  } = useResource<Config>('config', true);
-
-  const { mutate: addSignRule, error: addSignRuleError } = usePost(
-    `accounts/${nickname}/signrules`,
+  const [editingRule, setEditingRule] = useState<SignRule | undefined>(
+    undefined,
   );
+  const [isAddEditRuleModalOpen, setIsAddEditRuleModalOpen] = useState(false);
+
+  const [config, setConfig] = useState<Config>();
+  const [isLoading, setIsLoading] = useState(true);
+
+  const msWallet = useMemo(() => new MassaStationWallet(), []);
```

Comment:
Do not initialize a new MS wallet provider here.
Change the useProvider hook in `useProvider.tsx` to save the wallet object in state and do:
`const { wallet } = useProvider();`